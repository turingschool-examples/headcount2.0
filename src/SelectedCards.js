import React from 'react';
import Card from './Card';
import './SelectedCards.css';
import DistrictRepository from './helper.js';
import sampleData from './data/kindergartners_in_full_day_program';
import PropTypes from 'prop-types';

const district = new DistrictRepository(sampleData);

const SelectedCards = (props) => {
  let cardObjectList = Array.from(props.data);
  const districtCards = cardObjectList.reduce((districtCards, cardObject, index) => {
    if (props.selectedCards.includes(props.data[index].location)){
      districtCards.push(<Card key={index} {...props.data[index]} selectCard={props.selectCard} selectedCards={props.selectedCards}/>);
    }
    return districtCards;
  }, []);

  let district0;
  let district1;
  let districtsAverage;

  if (props.selectedCards[0] === props.selectedCards[1]) {
    props.selectedCards.shift();
    props.selectedCards.shift();
    districtCards.pop();
  }

  if (districtCards.length >= 1 && props.selectedCards.includes(districtCards[0].props.location)) {
    districtsAverage = <h2>{props.selectedCards[0]} {':'} {district.findAverage(districtCards[0].props.location)}</h2>;
  }

  if (districtCards.length === 2) {
    district0 = <h3>{props.selectedCards[1]} {':'} {district.findAverage(districtCards[0].props.location)}</h3>;
    district1 = <h3>{props.selectedCards[0]} {':'} {district.findAverage(districtCards[1].props.location)}</h3>;
    districtsAverage = <h2>{"<------ "} {district.compareDistrictAverages(districtCards[0].props.location, districtCards[1].props.location).compared} { " ------>"}</h2>;
  }

  return (
    <div className='selected-cards'>
      <p className='info-text'>Selected Comparisons</p>
      { districtCards[0] }
      <div className='compare-cards'>
        {district0}
        {districtsAverage}
        {district1}
      </div>
      { districtCards[1] }
    </div>
  );
};

SelectedCards.propTypes = {
  data: PropTypes.array,
  selectCard: PropTypes.func,
  selectedCards: PropTypes.array
}

export default SelectedCards;
