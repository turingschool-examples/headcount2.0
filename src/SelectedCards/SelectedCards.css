.selected-cards {
  color: rgb(255, 222, 173);
  display: flex;
  justify-content: space-between;
  max-width: 107rem;
  margin: 0 auto;
  height: 25rem;
  align-items: center;
  background-color: rgba(0, 255, 127, 0.2);
}

.selected-cards div {
  margin: 1rem;
}

.info-text {
  align-self: start;
    position: absolute;
    margin: 0rem;
    color: rgb(0, 255, 127);
}

.compare-cards {
  color: white;
  font-size: 1.5rem;
  text-align: center;
  margin: 0 auto;
  width: 30rem;
  height: 15rem;
}

@media only screen and (max-width: 1100px) {
  .selected-cards {
    width: 80vw;
    flex-direction: column;
    height: 50rem;
    margin: 5rem auto;
    padding: 1rem;
  }

  .info-text {
    margin: -1rem;
  }
}