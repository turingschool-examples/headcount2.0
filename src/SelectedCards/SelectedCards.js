import React from 'react';
import Card from '../Card/Card';
import './SelectedCards.css';
import DistrictRepository from '../helper.js';
import sampleData from '../data/kindergartners_in_full_day_program';
import PropTypes from 'prop-types';

const district = new DistrictRepository(sampleData);

const SelectedCards = (props) => {
  let cardList = Array.from(props.cleanData);
  const allCards = cardList.reduce((allCards, card, index) => {
    if (props.selectedCards.includes(props.cleanData[index].location)){
      allCards.push(
        <Card 
          key={index}
          {...props.cleanData[index]}
          selectCard={props.selectCard}
          selectedCards={props.selectedCards}
        />
      );
    }
    return allCards;
  }, []);

  let district0;
  let district1;
  let districtsAverage;
  const pickedCard = props.selectedCards;
  
  if (pickedCard[0] === pickedCard[1]) {
    pickedCard.shift();
    pickedCard.shift();
    allCards.pop();
  }
  
  if (allCards.length >= 1 && pickedCard.includes(allCards[0].props.location)) {
    districtsAverage = 
      <h2>
        {pickedCard[0]}
        {' : '}
        {district.findAverage(allCards[0].props.location)}
      </h2>;
  }

  if (allCards.length === 2) {
    const location0 = allCards[0].props.location;
    const location1 = allCards[1].props.location;

    district0 =
      <h3>
        {pickedCard[1]}
        {' : '}
        {district.findAverage(location0)}
      </h3>;
    district1 =
      <h3>
        {pickedCard[0]}
        {' : '}
        {district.findAverage(location1)}
      </h3>;
    districtsAverage =
      <h2>
        {"<------ "}
        {district.compareDistrictAverages(location0, location1).compared}
        { " ------>"}
      </h2>;
  }

  return (
    <div className='selected-cards'>
      <p className='info-text'>Selected Comparisons</p>
      { allCards[0] }
      <div className='compare-cards'>
        {district0}
        {districtsAverage}
        {district1}
      </div>
      { allCards[1] }
    </div>
  );
};

SelectedCards.propTypes = {
  cleanData: PropTypes.array.isRequired,
  selectCard: PropTypes.func.isRequired,
  selectedCards: PropTypes.array.isRequired
};

export default SelectedCards;
