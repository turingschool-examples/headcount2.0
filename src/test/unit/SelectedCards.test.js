import React from 'react';
import { shallow } from 'enzyme';
import SelectedCards from '../../SelectedCards/SelectedCards.js';

describe('SelectedCards', () => {
  it('should match snapshot', () => {
    const cleanData = [{
      location: 'COLORADO',
      stats: [{2007: 1}]
    }];
    const selectedCard = ['COLORADO'];
    const selectedCards = shallow(<SelectedCards cleanData={cleanData} selectedCards={selectedCard} selectCard={jest.fn()}/>);

    expect(selectedCards).toMatchSnapshot();
  });
});
