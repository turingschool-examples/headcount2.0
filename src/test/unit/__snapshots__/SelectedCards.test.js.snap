// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedCards should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectedCards
    cleanData={
        Array [
            Object {
              "location": "COLORADO",
              "stats": Array [
                Object {
                  "2007": 1,
                },
              ],
            },
          ]
    }
    selectCard={[Function]}
    selectedCards={
        Array [
            "COLORADO",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="info-text"
>
          Selected Comparisons
</p>,
        <Card
          location="COLORADO"
          selectCard={[Function]}
          selectedCards={
                    Array [
                              "COLORADO",
                            ]
          }
          stats={
                    Array [
                              Object {
                                "2007": 1,
                              },
                            ]
          }
/>,
        <div
          className="compare-cards"
>
          <h2>
                    COLORADO
                     : 
                    0.53
          </h2>
</div>,
        undefined,
      ],
      "className": "selected-cards",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Selected Comparisons",
          "className": "info-text",
        },
        "ref": null,
        "rendered": "Selected Comparisons",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "location": "COLORADO",
          "selectCard": [Function],
          "selectedCards": Array [
            "COLORADO",
          ],
          "stats": Array [
            Object {
              "2007": 1,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <h2>
              COLORADO
               : 
              0.53
</h2>,
            undefined,
          ],
          "className": "compare-cards",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "COLORADO",
                " : ",
                0.53,
              ],
            },
            "ref": null,
            "rendered": Array [
              "COLORADO",
              " : ",
              0.53,
            ],
            "type": "h2",
          },
          undefined,
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="info-text"
>
            Selected Comparisons
</p>,
          <Card
            location="COLORADO"
            selectCard={[Function]}
            selectedCards={
                        Array [
                                    "COLORADO",
                                  ]
            }
            stats={
                        Array [
                                    Object {
                                      "2007": 1,
                                    },
                                  ]
            }
/>,
          <div
            className="compare-cards"
>
            <h2>
                        COLORADO
                         : 
                        0.53
            </h2>
</div>,
          undefined,
        ],
        "className": "selected-cards",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Selected Comparisons",
            "className": "info-text",
          },
          "ref": null,
          "rendered": "Selected Comparisons",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "location": "COLORADO",
            "selectCard": [Function],
            "selectedCards": Array [
              "COLORADO",
            ],
            "stats": Array [
              Object {
                "2007": 1,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <h2>
                COLORADO
                 : 
                0.53
</h2>,
              undefined,
            ],
            "className": "compare-cards",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "COLORADO",
                  " : ",
                  0.53,
                ],
              },
              "ref": null,
              "rendered": Array [
                "COLORADO",
                " : ",
                0.53,
              ],
              "type": "h2",
            },
            undefined,
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
